icechunk/src/format/manifest.rs

```diff
@@ -41,16 +41,29 @@ pub enum VirtualReferenceError {
     UnsupportedScheme(String),
     #[error("error parsing bucket name from virtual ref URL {0}")]
     CannotParseBucketName(String),
-    #[error("error fetching virtual reference {0}")]
-    FetchError(Box<dyn std::error::Error + Send + Sync>),
+    #[error("error fetching virtual reference")]
+    FetchError(#[source] Box<dyn std::error::Error + Send + Sync>),
     #[error("the checksum of the object owning the virtual chunk has changed ({0})")]
     ObjectModified(String),
     #[error("error retrieving virtual chunk, not enough data. Expected: ({expected}), available ({available})")]
     InvalidObjectSize { expected: u64, available: u64 },
-    #[error("error parsing virtual reference {0}")]
+    #[error("error parsing virtual reference")]
     OtherError(#[from] Box<dyn std::error::Error + Send + Sync>),
 }
 
+pub type VirtualReferenceError = ICError<VirtualReferenceErrorKind>;
```

Comment:
we use the new machinery in error.rs to create actual errors and propagate the spans